drivers/ccpp/noahmpdrv.F90

```diff
@@ -100,9 +119,309 @@ subroutine noahmpdrv_init(lsm, lsm_noahmp, me, isot, ivegsrc, &
 
         pores (:) = maxsmc (:)
         resid (:) = drysmc (:)
+        
+        if (present(Land_IAU_Control) .and. present(Land_IAU_Data) .and. present(Land_IAU_State)) then 
+          ! Initialize IAU for land--land_iau_control was set by host model
+          if (.not. Land_IAU_Control%do_land_iau) return
+          call land_iau_mod_init (Land_IAU_Control, Land_IAU_Data, Land_IAU_State, errmsg, errflg) 
+        endif
 
       end subroutine noahmpdrv_init
 
+!> \ingroup NoahMP_LSM
+!! \brief This subroutine is called before noahmpdrv_run 
+!!  to update states with iau increments, if available
+!! \section arg_table_noahmpdrv_timestep_init Argument Table
+!! \htmlinclude noahmpdrv_timestep_init.html
+!!
+subroutine noahmpdrv_timestep_init (itime, fhour, delt, km,  ncols,         &      
```

Comment:
@barlage @tsga I think this is a new routine. Does this need to be called by the component model? At this point, the component model is just calling init and run. What happens if we don't call from component. Maybe it is just called in the IAU case.

Reply:
I believe @tsga has this routine only doing IAU so it doesn't need to be called in the component model for now.

Reply:
> I believe @tsga has this routine only doing IAU so it doesn't need to be called in the component model for now.

Yes, this function (and noahmpdrv_finalize) is only called for IAU, so the component model just skips it.